### PR TITLE
add all missing implicit roles from HTML 5

### PR DESCRIPTION
**Pull request by dylanb**
Don't merge yet, need to add tests

### CODE REVIEW

**Comment by dylanb** [comment]
closes https://github.com/dequelabs/axe-core/issues/252

**Comment by dylanb** [comment]
@marcysutton @WilcoFiers this is now ready for review/merge

File: lib/commons/aria/index.js [hunk 1]
```diff
@@ -473,7 +483,8 @@ lookupTables.role = {
 			one: ['menuitem', 'menuitemradio', 'menuitemcheckbox']
 		},
 		nameFrom: ['author'],
-		context: null
+		context: null,
+		implicit: ['menu[type="context"]']
```
Comment (marcysutton) [inline lib/commons/aria/index.js]: Huh, I didn't know you could specify implicit ARIA roles that way (with an attribute selector).

File: test/checks/aria/allowed-attr.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ describe('aria-allowed-attr', function () {
 	});
 
 	it('should return true if there is no role', function () {
-		var node = document.createElement('div');
+		var node = document.createElement('picture');
```
Comment (marcysutton) [inline test/checks/aria/allowed-attr.js]: After following https://github.com/w3c/html-aria/issues/52, it looks like `div` has no corresponding role (where I assumed it was `group`). Is that related?

File: test/mock/frames/recursive.html [hunk 3]
```diff
@@ -21,6 +21,6 @@
 </head>
 <body>
 	<h1>Recursive iframe</h1>
-	<iframe src="recursive.html">
+	<iframe src="nested1.html">
```
Comment (marcysutton) [inline test/mock/frames/recursive.html]: What is the nature of this change? It would help to know as I'm learning more about recursive iframes.

Comment (dylanb) [inline test/mock/frames/recursive.html]: `recursive.html` was infinitely including iframes. The infinite recursion was not required for that test to do what it was doing, so I changed it to a single iframe.
